packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/template-parts.js

```diff
@@ -21,14 +21,34 @@ import { MENU_ROOT, MENU_TEMPLATE_PARTS } from '../constants';
 
 export default function TemplatePartsMenu() {
 	const templateParts = useSelect( ( select ) => {
-		return select( 'core' ).getEntityRecords(
+		const publishedTemplateParts = select( 'core' ).getEntityRecords(
 			'postType',
 			'wp_template_part',
 			{
-				status: [ 'publish', 'auto-draft' ],
+				status: [ 'publish' ],
 				per_page: -1,
 			}
 		);
+		const currentTheme = select( 'core' ).getCurrentTheme()?.stylesheet;
+		const themeTemplateParts = select( 'core' ).getEntityRecords(
+			'postType',
+			'wp_template_part',
+			{
+				status: [ 'auto-draft' ],
+				per_page: -1,
+				theme: currentTheme,
+			}
+		);
+		// Results above can be 'null' in early steps of editor loading,
+		// so we must check before using the spread operator to combine them.
+		const combinedTemplateParts = [];
+		if ( publishedTemplateParts ) {
+			combinedTemplateParts.push( ...publishedTemplateParts );
```

Comment:
IMO, it would be clearer to use nullish coalescing if the query returns null, and then just rely on it being an array here:

```js
const publishedTemplateParts = select( 'core' ).getEntityRecords(
	'postType',
	'wp_template_part',
	{
		status: [ 'publish', 'auto-draft' ],
		status: [ 'publish' ],
		per_page: -1,
	}
) ?? [];

// same for themeTemplateParts 

return publishedTemplateParts.concat( themeTemplateParts ); // or spread
```

Reply:
oh, right!  🤦‍♀️   that would be much better.

Reply:
I suppose `??` and `||` are rather equivalent in this case.  `0` and `''` arent iterable either, but we would never reasonably expect them to be returned by the entity records either.  either way itl clean this up a bit.